tbxforms/static/sass/govuk/components/checkboxes/_index.scss

```diff
@@ -0,0 +1,309 @@
+@import "../error-message/index";
+@import "../fieldset/index";
+@import "../hint/index";
+@import "../label/index";
+
+@include tbxforms-exports("govuk/component/checkboxes") {
+  $tbxforms-touch-target-gutter: 4px;
+  $tbxforms-checkboxes-size: 40px;
+  $tbxforms-touch-target-size: calc($tbxforms-checkboxes-size + $tbxforms-touch-target-gutter);
+  $tbxforms-small-checkboxes-size: 24px;
+  $tbxforms-checkboxes-label-padding-left-right: tbxforms-spacing(3);
+  $tbxforms-checkbox-check-horizontal-position: 10px;
+
+  .tbxforms-checkboxes__item {
+    display: flex;
+    flex-wrap: wrap;
+    position: relative;
+    margin-bottom: tbxforms-spacing(2);
+  }
+
+  .tbxforms-checkboxes__item:last-child,
+  .tbxforms-checkboxes__item:last-of-type {
+    margin-bottom: 0;
+  }
+
+  .tbxforms-checkboxes__input {
+    // Allow the input to sit above the label, enabling its proper detection
+    // when exploring by touch or using automation tools like Selenium
+    z-index: 1;
+    width: $tbxforms-touch-target-size;
+    height: $tbxforms-touch-target-size;
+    margin: 0;
+    opacity: 0;
+    cursor: pointer;
+  }
+
+  .tbxforms-checkboxes__label {
+    align-self: center;
+
+    // Ensure that the width of the label is never more than the width of the
+    // container minus the input width minus the padding on either side of
+    // the label. This prevents the label from going onto the next line due to
+    // __item using flex-wrap because we want hints on a separate line.
+    max-width: calc(100% - #{(($tbxforms-checkboxes-label-padding-left-right * 2) + $tbxforms-touch-target-size)});
+    margin-bottom: 0;
+    padding: calc(tbxforms-spacing(1) + $tbxforms-border-width-form-element) tbxforms-spacing(3);
+    cursor: pointer;
+    // remove 300ms pause on mobile
+    touch-action: manipulation;
+  }
+
+  // [ ] Check box
+  .tbxforms-checkboxes__label::before {
```

Comment:
Personally I usually find these custom checkbox styles overkill for projects I'm working on. I'd like to have an option for a completely plain checkbox which can be customised as necessary.

Reply:
Interesting, what is overkill about it? GOV.UK offer a smaller variation which more closely resembles a default checkbox: https://design-system.service.gov.uk/components/checkboxes/#smaller-checkboxes

I'm hesitant to introduce a way to deviate from this, as this is the primary reason tbxforms exists - to avoid creating bespoke solutions for patterns that already have best practices. While heavy customisations may be needed, as you suggest, this feels like something best placed in project code and maintained as debt when upgrading tbxforms.

Reply:
It seems like a *lot* of css, and uses some really exact px values, and completely overrides the browser default checkbox styles. By contrast, here is some pretty simple css to remove the default checkbox icon and add a custom one: https://github.com/torchbox/torchbox.com/blob/main/tbx/static_src/sass/components/_mailchimp.scss#L62-L76 - and what if you just want to use the browser default style for the icon and just customise the border? My argument is that GDS seems overly heavy here, at least from glancing at this code.

Reply:
I agree the checkmark bit could be simplified by using an svg bg. Sometimes even just using a the default input with `accent-color` is enough styling for checkboxes if you don't need borders or a custom checkmark.

 https://developer.mozilla.org/en-US/docs/Web/CSS/accent-color

Reply:
Thanks both! However, as this is vendor code copied from GDS: https://github.com/alphagov/govuk-frontend/blob/main/packages/govuk-frontend/src/govuk/components/checkboxes/_index.scss and not something this MR introduces as a change, this isn't the right place for this discussion.

We aren't in a position to change this without incurring the debt to maintain it ourselves - something we don't do anywhere else.

If you feel strongly that the styles could be improved, [opening an issue on govuk-frontend](https://github.com/alphagov/govuk-frontend/issues) would be the best place.

If you feel strongly that tbxforms should deviate from GDS and define its own checkbox styles, then [opening an issue on tbxforms](https://github.com/torchbox/tbxforms/issues) would be the best place.